modules/backend/clinic/staff/doctor/doctor.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "i_visit_doctor.hpp"
+
+class Doctor : public iDoctor
+{
+    PersonalData personal_data_;
+    Specialization specialization_;
+
+  public:
+    Doctor(const PersonalData& personal_data, const Specialization specialization) : personal_data_{personal_data}, specialization_{specialization}
+    {
+    }
+    PersonalData& GetPersonalData() override
+    {
+      return personal_data_;
+    }
+    Specialization GetSpecialization() const override
+    {
+      return specialization_;
```

Comment:
as long as getPersonalData() and getSpecialization() are simple getters, I think it is best to keep their implementation in base class. They might be virtual or not. Even if it makes us resing from abstractness of base class. Until we want special implementation in some derived class it is maybe better to keep them non virtual even.

I think that Doctor should then implement funtions from other derived interfaces (if they cannot be implemented in base classes as well) or it's specific functions.

As long as no other interfaces are inherited or specific implementation appears, Doctor might be itself a base or final class.